renku/api/repository.py

```diff
@@ -227,7 +231,7 @@ def find_previous_commit(self, paths, revision='HEAD'):
                 )
             )
 
-        return file_commits[0]
+        return file_commits[-1 if first else 0]
```

Comment:
This will explode if file_commits is []. We should add a sentinel and raise proper renku exception in this case.

Reply:
it's raised a few lines above - the code only gets here if `file_commits` is not empty